samza-yarn/src/main/scala/org/apache/samza/job/yarn/YarnJob.scala

```diff
@@ -130,14 +130,14 @@ class YarnJob(config: Config, hadoopConfig: Configuration) extends StreamJob {
       Thread.sleep(1000)
     }
 
-    Running
+    getStatus
   }
 
   def getStatus: ApplicationStatus = {
     getAppId match {
       case Some(appId) =>
         logger.info("Getting status for applicationId %s" format appId)
-        client.status(appId).getOrElse(null)
+        client.status(appId).get // Should throw if None
```

Comment:
Should this log something before throwing?

Reply:
There's some logging in the client.status method. None value should never be returned, so I'll make that an exception, as it would represent a bug that users can't really do anything about.